### DISCUSSION TITLE
Using your profiles in Gentoo

### DISCUSSION THREAD

**Discussion by aahsnr-configs** [roddhjav/apparmor.d#486] [Q&A]
Hi, I looked over your wiki on how to install these profiles. I only looked at the arch version but guessing that you have a wiki for Arch Linux, that your profiles have been tested in arch. I looked at the PKGBUILD and finding that you are the aur maintainer as well, do you think I could replicate the steps in the PKGBUILD manually in Gentoo? The steps are basically make and make install except I don't understand the step with make: "make DISTRIBUTION=arch". What would specifying the type of distribution achieve?

**Comment by roddhjav** [comment]
For your initial test, you can build with `make DISTRIBUTION=arch`, later we would have to add *real* support in it.

This settings is used to:
- Set profile flags and ignore some profile https://apparmor.pujol.io/development/install/#profile-flags
- Set some distributions specificities such as the ABI version to support, or some package conflict.
- Resolve directives: https://apparmor.pujol.io/development/directives/#only-exclude

**Comment by maybebyte** [comment]
I did some light testing today on my Gentoo desktop targeting the 4.0 ABI. I'm able to boot and start my graphical environment (sway) without issue. There are some breakages, but they shouldn't be too hard to sort out when the time comes.

However, I think there will need to be a little work done on the Gentoo side before it makes sense to start working on integration here. The most immediate issue is that the relevant packages need to be brought up to date, to 4.1. Here is a list for anyone that's curious or wanting to help out:

- [`sec-policy/apparmor-profiles`](https://gitweb.gentoo.org/repo/gentoo.git/tree/sec-policy/apparmor-profiles) (this one has a [stalled PR open](https://github.com/gentoo/gentoo/pull/37834))
- [`sys-libs/libapparmor`](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-libs/libapparmor)
- [`sys-apps/apparmor-utils`](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-apps/apparmor-utils)
- [`sys-apps/apparmor`](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-apps/apparmor)

The latest stable for all of them in Gentoo is 3.1.4, and the latest testing is 4.0.3 for all of them except `sec-policy/apparmor-profiles`. I see in the announcements here that [the ABI for 3.x and 4.0 is getting deprecated](https://github.com/roddhjav/apparmor.d/discussions/738), so having up to date packages is even more of a consideration.

Later on, I think ideally there would be official support for Gentoo in the code here, and an ebuild in GURU (our version of the AUR, more or less). Not sure how that would look yet.

Also, I'd be happy to help out where I can. I'll probably take a look at updating the packages in Gentoo when I get a chance (assuming no one gets to it before I do) and go from there.

Apparmor version: 4.0.3
Kernel: 6.12.21 (custom kernel)
Init system: systemd (I think systemd is the only init system apparmor.d supports anyway)
Architecture: amd64

**Reply by roddhjav** [reply]
Nice, yes, updating apparmor ebuild for Gentoo is the priority. I am ok to test it and add it as supported distribution once apparmor get updated.
